sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionContext.scala

```diff
@@ -20,19 +20,23 @@ package org.apache.spark.sql.scripting
 import scala.collection.mutable.ListBuffer
 
 import org.apache.spark.SparkException
+import org.apache.spark.sql.scripting.SqlScriptingFrameType.SqlScriptingFrameType
 
 /**
  * SQL scripting execution context - keeps track of the current execution state.
  */
 class SqlScriptingExecutionContext {
   // List of frames that are currently active.
-  val frames: ListBuffer[SqlScriptingExecutionFrame] = ListBuffer.empty
+  private[scripting] val frames: ListBuffer[SqlScriptingExecutionFrame] = ListBuffer.empty
```

Comment:
not related to this PR, but can we add more comments to explain the concepts (frame, scope)? we should also try to map them to the user-facing concepts: single statement, compound statement (BEGIN END), error hander statement, nested stored procedure, etc.